util/Migrator/DbMigrator.cs

```diff
@@ -12,39 +12,34 @@ public class DbMigrator
 {
     private readonly string _connectionString;
     private readonly ILogger<DbMigrator> _logger;
-    private readonly string _masterConnectionString;
 
-    public DbMigrator(string connectionString, ILogger<DbMigrator> logger)
+    public DbMigrator(string connectionString)
     {
         _connectionString = connectionString;
-        _logger = logger;
-        _masterConnectionString = new SqlConnectionStringBuilder(connectionString)
-        {
-            InitialCatalog = "master"
-        }.ConnectionString;
+        _logger = CreateLogger();
     }
 
     public bool MigrateMsSqlDatabaseWithRetries(bool enableLogging = true,
         bool repeatable = false,
         string folderName = MigratorConstants.DefaultMigrationsFolderName,
-        CancellationToken cancellationToken = default(CancellationToken))
+        CancellationToken cancellationToken = default)
     {
         var attempt = 1;
-
         while (attempt < 10)
         {
             try
             {
+                PrepareDatabase(cancellationToken);
```

Comment:
ℹ️ Separation of "preparation" tasks from the actual migration. The tasks in here really don't have to run any time but once, but not going after that here.